nemo/collections/asr/models/sortformer_diar_models.py

```diff
@@ -0,0 +1,578 @@
+from nemo.collections.asr.parts.preprocessing.perturb import process_augmentations
+from nemo.collections.asr.parts.utils.asr_multispeaker_utils import get_ats_targets, get_pil_targets
+from nemo.collections.common.data.lhotse import get_lhotse_dataloader_from_config
+from nemo.core.classes import ModelPT
+from nemo.core.classes.common import PretrainedModelInfo
+from nemo.core.neural_types import AudioSignal, LengthsType, NeuralType
+from nemo.core.neural_types.elements import ProbsType
+from nemo.utils import logging
+
+__all__ = ['SortformerEncLabelModel']
+
+
+class SortformerEncLabelModel(ModelPT, ExportableEncDecModel):
+    """
+    Encoder class for Sortformer diarization model.
+    Model class creates training, validation methods for setting up data performing model forward pass.
+
+    This model class expects config dict for:
+        * preprocessor
+        * Transformer Encoder
+        * FastConformer Encoder
+        * Sortformer Modules
+    """
+
+    @classmethod
+    def list_available_models(cls) -> List[PretrainedModelInfo]:
+        """
+        This method returns a list of pre-trained model which can be instantiated directly from NVIDIA's NGC cloud.
+
+        Returns:
+            List of available pre-trained models.
+        """
+        result = []
+        return result
+
+    def __init__(self, cfg: DictConfig, trainer: Trainer = None):
+        """
+        Initialize an Sortformer Diarizer model and a pretrained NEST encoder.
+        In this init function, training and validation datasets are prepared.
+        """
+        random.seed(42)
+        self._trainer = trainer if trainer else None
+        self._cfg = cfg
+
+        if self._trainer:
+            self.world_size = trainer.num_nodes * trainer.num_devices
+        else:
+            self.world_size = 1
+
+        if self._trainer is not None and self._cfg.get('augmentor', None) is not None:
+            self.augmentor = process_augmentations(self._cfg.augmentor)
+        else:
+            self.augmentor = None
+        super().__init__(cfg=self._cfg, trainer=trainer)
+        self.preprocessor = SortformerEncLabelModel.from_config_dict(self._cfg.preprocessor)
+
+        if hasattr(self._cfg, 'spec_augment') and self._cfg.spec_augment is not None:
+            self.spec_augmentation = SortformerEncLabelModel.from_config_dict(self._cfg.spec_augment)
+        else:
+            self.spec_augmentation = None
+
+        self.encoder = SortformerEncLabelModel.from_config_dict(self._cfg.encoder)
+        self.sortformer_modules = SortformerEncLabelModel.from_config_dict(self._cfg.sortformer_modules)
+        self.transformer_encoder = SortformerEncLabelModel.from_config_dict(self._cfg.transformer_encoder)
+        self._init_loss_weights()
+
+        self.eps = 1e-3
+        self.loss = instantiate(self._cfg.loss)
+
+        self.streaming_mode = self._cfg.get("streaming_mode", False)
+        self.save_hyperparameters("cfg")
```

Comment:
isn;t this automatically done when calling super init?

Reply:
As far as I remember, without this line, there was an issue of yaml file inside the checkpoint - showing missing parameters